src/mnode/ticket-processor.cpp

```diff
@@ -685,39 +696,61 @@ std::string CPastelTicketProcessor::ListFilterSellTickets(const short filter) co
                     return true;
             }
             return false;
-        });
+        }, checkConfirmation);
 }
 
-// 1 - expired;    2 - sold
-std::string CPastelTicketProcessor::ListFilterBuyTickets(const short filter) const
+// 0 - all, 1 - expired;    2 - sold
+std::string CPastelTicketProcessor::ListFilterBuyTickets(const short filter, const std::string& pastelID) const
 {
+    bool checkConfirmation = true;
```

Comment:
probable just write:
const bool checkConfirmation{filter > 0};
and remove
"else
            checkConfirmation = false;"